woocommerce/woocommerce-ios
# Shipping Labels M1: create new entities in storage layer

Fixes #3113 

The PR size is ~830 diffs, but ~592 diffs are on the new model version 38 and bunch of auto-generated code for Core Data subclasses!

## Changes

- Added model version 38 with changes in the following commits: https://github.com/woocommerce/woocommerce-ios/commit/14047aa272263c4bb2538cd8a5e10a9c6c056b93, https://github.com/woocommerce/woocommerce-ios/commit/d7a0d27f85fb5be4d047c96ff2f154d3255ca806, https://github.com/woocommerce/woocommerce-ios/commit/a9341bfca00b52f8e830ce2d994ef969f3df07cd, and https://github.com/woocommerce/woocommerce-ios/pull/3168/commits/759695fe4f2cbc52edd1bace0a32b46f97f57d1d - the last three commits were made after initial UI integration and PR feedback.
- Model version 38 has 4 new entities: `ShippingLabel`, `ShippingLabelAddress`, `ShippingLabelRefund`, and `ShippingLabelSettings`. Generated subclasses for these 4 entities.
- Added a test case to `MigrationTests` which verifies that these 4 new entities can be created in storage layer in model version 38.

## Testing

Please sanity check on migration:

- Launch the app from `develop` or any other branch without changes in this PR, and log in if necessary
- Launch the app from PR branch --> the app should run normally without crashing

Update release notes:
